syntax = "proto3";

service ChatService {
    rpc RegisterUser (RegisterRequest) returns (RegisterResponse);
    rpc SendMessage (SendMessageRequest) returns (SendMessageResponse);
    rpc ReceiveMessage (ReceiveMessageRequest) returns (ReceiveMessageResponse);
}

message RegisterRequest {
    string username = 1;
    string ip = 2;
    int32 port = 3;
}

message RegisterResponse {
    string status = 1;
}

message SendMessageRequest {
    string sender = 1;
    string receiver = 2;
    string message = 3;
}

message SendMessageResponse {
    string status = 1;
}

message ReceiveMessageRequest {
    string sender = 1;
    string message = 2;
}

message ReceiveMessageResponse {
    string status = 1;
}